src/main/kotlin/it/pagopa/ecommerce/cdc/services/TimestampRedisTemplate.kt

```diff
@@ -0,0 +1,33 @@
+package it.pagopa.ecommerce.cdc.services
+
+import java.time.Duration
+import java.time.Instant
+import java.util.*
+import kotlin.text.format
+import org.springframework.data.redis.core.RedisTemplate
+import org.springframework.data.redis.core.ValueOperations
+import org.springframework.stereotype.Component
+
+// extension function for indexed accessor syntax with TTL
+// this is to avoid sonar warning on rule kotlin:S6518
+// https://next.sonarqube.com/sonarqube/coding_rules?open=kotlin%3AS6518&rule_key=kotlin%3AS6518
+operator fun ValueOperations<String, Instant>.set(key: String, ttl: Duration, value: Instant) {
+    set(key, value, ttl)
+}
+
+@Component
+class TimestampRedisTemplate(private val redisTemplate: RedisTemplate<String, Instant>) {
+    fun save(keyspace: String, cdcTarget: String, instant: Instant, ttl: Duration) {
+        redisTemplate.opsForValue()[compoundKeyWithKeyspace(keyspace, cdcTarget), ttl] = instant
+    }
```

Comment:
we can avoid create a custom function extension and use index accessor like this
```suggestion
@Component
class TimestampRedisTemplate(private val redisTemplate: RedisTemplate<String, Instant>) {
    fun save(keyspace: String, cdcTarget: String, instant: Instant, ttl: Duration) {
        redisTemplate.opsForValue()[compoundKeyWithKeyspace(keyspace, cdcTarget), instant] = ttl
    }
 both calls the same java set methods
```